### PR TITLE
ocamlformat

### PR DESCRIPTION
**Pull request by cknitt**
As discussed with @cristianoc, add ocamlformat to the project, but do not reformat files or enforce formatting in CI yet.

### CODE REVIEW

File: .ocamlformat [hunk 1]
```diff
@@ -0,0 +1,6 @@
+profile = default
+version = 0.22.4
+
```
Comment (cristianoc) [inline .ocamlformat:3]: See: https://github.com/rescript-lang/rescript-vscode/blob/master/analysis/.ocamlformat
There's nothing set in stone in that file, just that's been used so far.
Seems like a good idea to converge to one, and use it in all places.

**Comment by cristianoc** [comment]
Looking good.
Just need to make sure there are no big outstanding PRs at merge time.

File: .ocamlformat (hunk 1 above)
Comment (cknitt) [reply .ocamlformat:3]: I have taken over the additional settings from there now. I am trying to minimize the changes compared to the existing code style.

I am wondering if it is possible to get rid of the space before the `:` in

```ocaml
val loc : t -> Location.t
```

Haven't found the setting for that yet (only for `let` bindings).

File: src/res_utf8.ml [hunk 2]
```diff
@@ -24,118 +24,365 @@ let h4 = 0b1111_0000
...
-  0; 0; 2; 2;2; 2; 2; 2;2; 2; 2; 2;2; 2; 2; 2;
-   2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2;
-   3; 4; 4; 4; 4; 4; 4; 4; 4; 4; 4; 4; 4; 5; 4; 4;
-  6; 7; 7 ;7; 8; 0; 0; 0; 0; 0; 0; 0; 0; 0; 0; 0;
-|]
+let categories =
+  [|
+    1;
+    1;
+    1;
```
Comment (cknitt) [inline src/res_utf8.ml:53]: @cristianoc This is not ideal. Probably should put the tables into a separate file that is excluded from formatting.

Comment (cknitt) [reply src/res_utf8.ml:53]: Ah there is an easier way with `[@@ocamlformat "disable"]`.

**Comment by cknitt** [comment]
Now also checking for formatting in CI.

**Review by cristianoc — APPROVED** [review]
Great, thanks for doing this!

**Pull request merged**